cloud/scope/object_storage_bucket.go

```diff
@@ -161,6 +162,50 @@ func (s *ObjectStorageBucketScope) ApplyAccessKeySecret(ctx context.Context, key
 	return nil
 }
 
+// getAccessKeysFromSecret gets the access keys id for the OBJ buckets from a secret
+func (s *ObjectStorageBucketScope) GetAccessKeysFromSecret(ctx context.Context, secretName string, secretNamespace string) ([2]float64, error) {
+
+	var newKeys [2]float64
+	// Delete the access keys.
+	objkey := client.ObjectKey{
+		Namespace: secretNamespace,
+		Name:      secretName,
+	}
+	var secret corev1.Secret
+	if err := s.client.Get(ctx, objkey, &secret); err != nil {
+		if apierrors.IsNotFound(err); err != nil {
+			return newKeys, err
+		}
+	}
+	for i, e := range []struct {
+		permission string
+		suffix     string
+	}{
+		{"read_write", "rw"},
+		{"read_only", "ro"},
+	} {
+		secretDataForKey, isset := secret.Data[e.permission]
+		if !isset {
+			return newKeys, fmt.Errorf("secret %s missing data field: %s", secretName, e.permission)
+		}
+		decodedSecretDataForKey := string(secretDataForKey)
+
+		var jsonMap map[string]interface{}
+		json.Unmarshal([]byte(string(decodedSecretDataForKey)), &jsonMap)
+
+		accessKeyID, ok := jsonMap["id"]
+		if !ok {
+			err := fmt.Errorf("obj bucket key not found")
+			return newKeys, err
+		}
+		key := accessKeyID.(float64)
```

Comment:
Isn't the `id` field just an `int`? We can unmarshal into a `linodego.ObjectStorageKey` to not have to deal with `any` types.